### ISSUE TITLE
Graphical Issues with MultiMC Not Present with Vanilla Launcher (No Mods)

### ISSUE THREAD

**Issue by Cyndan** [MultiMC/MultiMC5#2319]
System Information
-----------------------------
MultiMC version: 0.6.2-1222
Operating System: Windows 8.1 64bit
Intel Core i7-4720HQ CPU @ 2.60 GHz
16.0 GB RAM (4GB dedicated)
Built in Intel HD Graphics 4600 / NVIDIA GeForce GTX 960M
   (As a side note on this, if need be I can manually check or set which graphics card is used, and both the vanilla launcher and MultiMC use the NVIDIA card)

All within an Acer Aspire V Nitro laptop, purchased three years ago.

Summary of the issue or suggestion:
----------------------------------------------
     While the base vanilla launcher for Minecraft at version 1.12.2 works perfectly fine and remarkably well, even, when running the same vanilla Minecraft version 1.12.2 through MultiMC, I come across some graphical errors that at times crash the client. 
     If VBO's remain enabled, the game crashes as soon as chunks are loaded, saved, and the game is left unpaused for a single frame. When this happens, there is either no error given (just the client closing or crashing with a "[Minecraft] has stopped responding" message as soon as The End's chunk is saved with exit code 255), or there is an "EXCEPTION_ACCESS_VIOLATION" and a line about being unable to write core dump. The difference between these two outcomes appear to be allocated memory; when below 3~4GB of RAM, the access violation error occurs with exit code -1073740940 or 1; I feel like with testing this I've seen just about every error code there is.
     If VBO's are disabled, the game will at the very least run and not crash immediately. However, the screen flickers with a whitish, bluish, sky-colored plane every now and again that is made worse by moving. In addition, when holding items, a black lattice-like texture sometimes appears over the item in question, among other graphical UI errors such as the item briefly disappearing. I've also noticed that on occasion, elements of the pause menu do not fully load.

What should happen:
------------------------------
Ideally, the game should run similarly to how it does with the vanilla launcher.

Steps to reproduce the issue (Add more if needed):
-------------------------------------------------------------
For a "true crash":
1. Allocate below 3GB of RAM
2. Turn VBO's on in the Video Options within the client
3. Attempt to load or create a world

For graphical errors:
1. Allocate any amount of RAM
2. Turn VBO's off in the Video Options within the client
3. Load or create a world

Suspected cause:
---------------------------
¯\_(ツ)_/¯

Logs/Screenshots:
----------------------------
[01]: https://paste.ee/p/g0AKi (Log. VBO's on, lower RAM)
[02]: https://paste.ee/p/riCIw (Log. VBO's on, higher RAM)
[03]: https://paste.ee/p/b7Zw1 (Crash report. VBO's on, lower RAM)
[04]: https://youtu.be/Kq6BgCaTJzQ (Video recording of what I see, with VBO's both on and off.)

Additional Info:
---------------------------
This information is all for barely-touched or brand new worlds. I'm noticing that I get different errors with VBO's if I've played on a world for more than a minute or two, namely regarding a warning about an entity "minecraft:bat" already existing and I'm honestly too exhausted to get into that can of worms right now, but will if it becomes more relevant.

I have attempted, while troubleshooting for a total of four hours so far, in no particular order:
-Updating and reinstalling my graphics card driver
-Updating and reinstalling Java
-Using a slightly older version of Java
-Deleting old versions of Java
-Making sure only one instance of Java is running
-Renaming MultiMC to "Minecraft[.exe]"
-Allocating more or less RAM
-Deleting any trace of Minecraft from my folders, including roaming
-Reinstalling MultiMC. Twice.
-Fiddling with video settings
-Using integrated graphics instead of NVIDIA graphics
-Trying older versions of Minecraft, as far back as 1.6.4
-Renaming a particular folder that contained a space character to no longer have said space
-Slamming my head into my desk

Things I have not attempted:
-Manually reinstalling the Java Game Library file dll thingy, as it's a bit beyond my scope of knowledge (and as mentioned, vanilla Minecraft works perfectly fine, anyway)
-Multiplayer
-Optifine
-Playing for more than two minutes or any actual progression
-Crying

I know that I have played Minecraft using MultiMC before with mods installed on a friend's server; this would have been approximately...8-10 months ago, by my best estimation. I encountered these errors while testing a new modpack, and realized that I was still getting errors even without mods, leading me to this here ;w;

**Comment by peterix** [comment]
This *should be* between the game and your graphics drivers, but reality is often a lot more m messy.

- make absolutely sure that you are actually using the Nvidia gpu.
- disable any and all 'performance accelerators', overlays and basically everything that isn't the bare OS, bare gpu drivers and the game.

**Comment by Cyndan** [comment]
Okay, so unfortunately yesterday my router decided to fizzle out, so I wasn't able to be any sort of online until around 10pm. That said, I did figure this out after taking a look at your suggestions from my phone.

So what was happening was that although I was setting MultiMC to use my NVIDIA card, once it launched Minecraft, it uses a different instance of Java that defaults to my integrated graphics; I had to take a look at task manager to find that bit out. Setting that file to run NVIDIA instead fixed my issues!
I know that NVIDIA supports Minecraft, so it no doubt recognized the official launcher and set it automatically from there, but didn't recognize MultiMC's version, so that's my theory, anyhow.
